src/main/scala/com/cloudera/sparkts/TimeSeriesStatisticalTests.scala

```diff
@@ -246,4 +246,34 @@ object TimeSeriesStatisticalTests {
     }
     diffsSum / residsSum
   }
+  
+  /**
+   * Breusch-Godfrey test for serial correlation in a model
+   *
+   * @return The Breusch-Godfrey statistic and p value. The statistic asymptotically 
+   * follows an X^2 distribution with maxLag degrees of freedom, and provides a test for the null hypothesis
+   * of lack of serial correlation up to degree maxLag. We implement in the form of the
+   * Lagrange Multiplier test
+   * See http://en.wikipedia.org/wiki/Breusch%E2%80%93Godfrey_test for more information
+   * BG test is robust to various situations in which DWtest and durbin h statistic assumptions are broken
+   */
+  
+  def bgtest(ts: Vector[Double], factors: Matrix[Double], maxLag: Int): (Double, Double) = {
+      /* original regression model */
+      val orig_ols = new OLSMultipleLinearRegression()
+      val factor_arrays = Util.matToRowArrs(factors) /* X */
```

Comment:
Can you mention where these variable names are coming from? I assume the Wikipedia article?